### DISCUSSION TITLE
Overload resolution for FormattableString (bug?)

### DISCUSSION THREAD

**Discussion by rikimaru0345** [dotnet/csharplang#675] [Q&A] [answered]
Hello, I have three small questions.

1. Calling `L($"some {i} formatted string");` will use the `string` overload
instead of the `FormattableString` given those two methods:
```csharp
public static string L(FormattableString fs)
	=> string.Format(fs.Format, fs.GetArguments());

public static string L(string s)
	=> s;
```

2. When you just have the following (now an extension method), you cannot call it
like this `$"test {i} string".L();`

```csharp
public static string L(this FormattableString fs)
{
	return string.Format(fs.Format, fs.GetArguments());
}
```

Is the second one really supposed to not work, calling L(...) directly will work though.
As for the first one, since `FormattableString`is more rarely used, shouldn't it have higher priority?
It would make things a little easier.

3. Why isn't `FormattableString` not a struct? It always allocates, couldn't that have been prevented easily?

I hope someone can explain if this is just an unintended side-effect, or if there's any actual reasoning behind this behaviour.

**Comment by yaakov-h** [comment, marked as answer]
dotnet/roslyn#173
dotnet/roslyn#518
dotnet/roslyn#8097

**Comment by rikimaru0345** [comment]
I read the threads you linked.
Wow... that sucks.

99% of the users could live without being able to sub-class FormattableString (they can write a library or something), but bad performance can't be fixed. Really sad. Bad decisions.
